### ISSUE TITLE
SA1015 false positive on generic lambda return type declaration

### ISSUE THREAD

**Issue by rdeago** [DotNetAnalyzers/StyleCopAnalyzers#3655]
**Version 1.2.0-beta.435**

When declaring the return type of a lambda, and said return type is a generic type, warning SA1015 is incorrectly issued.

Simplest repro I could come up with:

```c#
using System;
using System.Collections.Generic;

//           +-- Warning SA1015: Closing generic bracket should not be followed by a space
//           |
//           v
Test(List<int> () => new());

void Test(Delegate x)
{
}
```

Of course, removing the space after the incriminated bracket causes warning SA1008 (Opening parenthesis should be preceded by a space) which is correct IMO.

This bug gets particularly annoying when using ASP.NET Core Minimal APIs and declaring a handler as returning one among two or more `IResult` implementations:

```c#
//                                                           +-- Warning SA1015
//                                                           |
//                                                           v
app.MapGet("/orders/{orderId}", Results<BadRequest, Ok<Order>> (int orderId) 
    => orderId > 999 ? TypedResults.BadRequest() : TypedResults.Ok(new Order(orderId)));
```

The above example code [comes straight from ASP.NET Core documentation](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis/responses?view=aspnetcore-7.0#resultstresult1-tresultn).

**Comment by bjornhellander** [comment]
Duplicate of #3624. This has already been fixed, but not yet released. Will be included in the next release.

**Comment by jizc** [comment]
@bjornhellander do you know if there is an estimate about when the next beta release will be? It's been over a year since the last one, and there seems to be many unreleased improvements. Is there some blocker preventing a release?

**Comment by bjornhellander** [comment]
The fix is relased in 1.2.0-beta.507, so I assume this can be closed now.
